cmd/nse/nse.go

```diff
@@ -0,0 +1,171 @@
+)
+
+const (
+	// clientConnectionTimeout defines time the client waits for establishing connection with the server
+	clientConnectionTimeout = time.Second * 60
+	// clientConnectionTimeout defines retry interval for establishing connection with the server
+	clientConnectionRetry = time.Second * 2
+)
+
+var (
+	clientSocketPath     = path.Join(nsmserver.SocketBaseDir, nsmserver.ServerSock)
+	clientSocketUserPath = flag.String("nsm-socket", "", "Location of NSM process client access socket")
+	nseSocketName        = flag.String("nse-socket", "nse.ligato.io.sock", "Name of NSE socket whcih will be used by NSM for Connection Request call")
+)
+
+func dial(ctx context.Context, unixSocketPath string) (*grpc.ClientConn, error) {
+	c, err := grpc.DialContext(ctx, unixSocketPath, grpc.WithInsecure(), grpc.WithBlock(),
+		grpc.WithDialer(func(addr string, timeout time.Duration) (net.Conn, error) {
+			return net.DialTimeout("unix", addr, timeout)
+		}),
+	)
+
+	return c, err
+}
+
+func main() {
+	flag.Parse()
+	flag.Set("logtostderr", "true")
+
+	// Checking if nsm client socket exists and of not crash NSE
+	clientSocket := clientSocketPath
+	if clientSocketUserPath != nil {
+		clientSocket = *clientSocketUserPath
+	}
+
+	if _, err := os.Stat(clientSocket); err != nil {
+		logrus.Errorf("nse: failure to access nsm socket at %s with error: %+v, exiting...", clientSocket, err)
+		os.Exit(1)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), clientConnectionTimeout)
+	defer cancel()
+	conn, err := dial(ctx, clientSocket)
+	if err != nil {
+		logrus.Errorf("nse: failure to communicate with the socket %s with error: %+v", clientSocket, err)
+		os.Exit(1)
+	}
+	defer conn.Close()
+	logrus.Infof("nse: connection to nsm server on socket: %s succeeded.", clientSocket)
+
+	// NSM socket path will be used to drop NSE socket for NSM's Connection request
+	nsePath, _ := filepath.Split(clientSocket)
+	if err := socketCleanup(path.Join(nsePath, *nseSocketName)); err != nil {
+		logrus.Errorf("nse: failure to cleanup stale sicket %s with error: %+v", path.Join(nsePath, *nseSocketName), err)
+		os.Exit(1)
+	}
+	nse, err := net.Listen("unix", path.Join(nsePath, *nseSocketName))
+	grpcServer := grpc.NewServer([]grpc.ServerOption{}...)
+
+	go func() {
+		if err := grpcServer.Serve(nse); err != nil {
+			logrus.Errorf("nse: failed to start grpc server on socket %s with error: %+v ", path.Join(nsePath, *nseSocketName), err)
+		}
+	}()
+	// Check if the socket of device plugin server is operation
+	if err := socketOperationCheck(path.Join(nsePath, *nseSocketName)); err != nil {
+		logrus.Errorf("nse: failure to communicate with the socket %s with error: %+v", path.Join(nsePath, *nseSocketName), err)
+		os.Exit(1)
+	}
+
+	// Ok, NSE server is ready and now the channel can be advertised to NSM
+	nsmClient := nsmconnect.NewClientConnectionClient(conn)
+
+	channel := netmesh.NetworkServiceChannel{
+		Metadata: &common.Metadata{
+			Name: "Channel-1",
```

Comment:
Why is -1 hard coded here?

Reply:
As I stated in PR description, it is just a simple nse I thought would be used in CI for testing purposes. (I will add integration nse into CI after I get ok from you and @edwarnicke and @mestery )
So I can add some parameters but again it is not meant to be a production level `nse`

Reply:
No problem with this then.

Reply:
This is a good idea to create one of these for testing, but I think we should add an explicit README.md or something to indicate this is essentially for testing purposes.